tests/datatypes/test_str.py

```diff
@@ -681,6 +681,12 @@ def __str__(self):
                 print(err)
             """)
 
+    def test_isnumeric(self):
+        self.assertCodeExecution("""
+        for str_ in ['123', '123.4', 'abc', '', ' ', 'ABCD', 'ABCD ', '12323445']:
```

Comment:
Could you also test some other cases like `'123.', '.12', '1A', 'A1'` to make sure the edge cases are handled correctly?